### PR TITLE
added linechart

### PR DESCRIPTION
**Pull request by zilehuda**
Added linechart module

### CODE REVIEW

**Comment by vimark1** [comment]
Hey @zilehuda, Thanks. This is looking good.  I just have 2 comments:
1. the chart data should be related to the current logged in user, not the top-scorers.  This is so that users can see their progress as they continue to use the site. 
2. We can put the chart component in the profile page, rather than a dedicated route for it

 Would you be able to address that?

**Comment by zilehuda** [comment]
> Hey @zilehuda, Thanks. This is looking good. I just have 2 comments:
> 
> 1. the chart data should be related to the current logged in user, not the top-scorers.  This is so that users can see their progress as they continue to use the site.
> 2. We can put the chart component in the profile page, rather than a dedicated route for it
> 
> Would you be able to address that?

SURE :D

**Comment by vimark1** [comment]
Awesome! btw the user data gets stored here:https://github.com/vimark1/typist/blob/1f8c2977ad6b24350fe65f3bc51de2e80d314bfc/src/Main/index.js#L199-L207

File: src/LineChartStat/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        super(props);
+        this.state = {
+            top10 : [
+                //{user: (user object), score: "12"}
+            ]
+        }
+    }
+    
+    componentDidMount() {
+        ReactGA.pageview('/scoreboard');
```
Comment (clayreimann) [inline src/LineChartStat/index.js]: This should be updated to reflect the new page.
```suggestion
        ReactGA.pageview('/chart');
```

**Comment by vimark1** [comment]
@clayreimann 
> The graph page here should be rolled in to the existing Scoreborad page

Yeah, we dont need another route for this, the graph should be added to the profile page, we'll keep all other stats there for now.  see my previous comment on https://github.com/vimark1/typist/pull/81#issuecomment-433826284

>  I also think that it should be a bar chart, not a line graph. This is not a time-series data set. If you wanted to use a line graph you should graph the user's previous scores.

Yes, the graph is for the user's previous score. The implementation in this PR here is different to what I had imagined. I clarified this in the same comment https://github.com/vimark1/typist/pull/81#issuecomment-433826284

**Comment by clayreimann** [comment]
Yeah, I don’t know why but your comments didn’t show up when I was looking at the PR

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: src/Header/index.js [hunk 2]
```diff
@@ -28,6 +28,7 @@ export default class Header extends Component {
       </div>
       <div className="menu options">
         <NavLink className="menu-item" to="/scoreboard">Scoreboard</NavLink>
+        <NavLink className="menu-item" to="/linechart">linechart</NavLink>
```
Comment (vimark1) [inline src/Header/index.js:31]:
```suggestion
        <NavLink className="menu-item" to="/chart">Chart</NavLink>
```

File: src/App/index.js [hunk 3]
```diff
@@ -30,6 +31,7 @@ export default class App extends Component {
           <Route exact path='/signin' component={Signin} />
           <Route exact path='/signup' component={Signup} />
           <Route exact path='/scoreboard' component={ScoreBoard} />
+          <Route exact path='/linechart' component={LineChartStat} />
```
Comment (vimark1) [inline src/App/index.js:34]:
```suggestion
          <Route exact path='/chart' component={LineChartStat} />
```

File: src/index.tsx [hunk 4]
```diff
@@ -48,6 +49,7 @@ firebase.auth().onAuthStateChanged(user => {
               <Route exact={true} path="/signin" component={Signin} />
               <Route exact={true} path="/signup" component={Signup} />
               <Route exact={true} path="/scoreboard" component={ScoreBoard} />
+              <Route exact path='/chart' component={LineChartStat} />
```
Comment (vimark1) [inline src/index.tsx:52]:
```suggestion
              <Route exact={true} path="/chart" component={LineChartStat} />
```

File: src/pages/LineChartStat/index.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import ReactGA from 'react-ga';
+import firebase from 'firebase/app';
+import LineChart from 'react-linechart';
+import '../../node_modules/react-linechart/dist/styles.css';
```
Comment (vimark1) [inline src/pages/LineChartStat/index.tsx:5]:
```suggestion
import '../../../node_modules/react-linechart/dist/styles.css';
```

**Comment by vimark1** [comment]
@clayreimann and @zilehuda thanks for your contribution on this PR, it has been merged now

**Pull request merged**